programs/token-wrapper/src/lib.rs

```diff
@@ -74,6 +87,116 @@ pub mod token_wrapper {
                 set_return_data(&return_data);
                 Ok(())
             }
+            CalleeProgram::TokenMetadata => {
+                let meta = Metadata::from_account_info(&mint.to_account_info())?;
```

Comment:
It's very clever to make the metadata into the "mint" for the transfer, since you're sure that it has the data. 

How is that meant to be communicated to users of the wrapper program though? Is it one of those "you just have to know" things? Is it possible to figure out that a mint is managed by the metadata program based on some qualities of the token mint?

Reply:
I don't think it's possible to figure out `mint` is actually owned by `token-metadata`, since you need to fetch ownership account's data to check that.

Reply:
Oh wait hold on! I found https://github.com/metaplex-foundation/metaplex-program-library/blob/master/token-metadata/program/ProgrammableNFTGuide.md, and if I'm reading it right, if the mint and freeze authority are a particular PDA, then you know it's managed by Metaplex!

Reply:
And what's pretty cool, this is an example that would require *two* calls to figure out the required accounts, since the first call would just give you *more* accounts, and the second would allow you to derive everything

Reply:
Honestly it might be better to just rename the `mint` account to `descriptor` or something more generic